tf_fastmri_data/preprocessing_utils/fourier/non_cartesian.py

```diff
@@ -26,13 +26,5 @@ def _crop_for_nufft(image, im_size):
 def nufft(nufft_ob, image, ktraj, image_size=None):
     forward_op = kbnufft_forward(nufft_ob._extract_nufft_interpob(), multiprocessing=True)
     shape = tf.shape(image)[-1]
-    if image_size is not None:
-        image_adapted = tf.cond(
-            tf.math.greater(shape, image_size[-1]),
-            lambda: _crop_for_nufft(image, image_size),
-            lambda: _pad_for_nufft(image, image_size),
-        )
-    else:
-        image_adapted = image
-    kspace = forward_op(image_adapted, ktraj)
+    kspace = forward_op(image, ktraj)
```

Comment:
Can you tell me again why you felt it was better to have the size adjustment inside the dataset rather than the nufft?

Reply:
Actually, I think I am rushing. Let me try to make code cleaner... There seems to be redundacy!

Reply:
I think in the current state you are just applying `adjust_image_size` twice, also once without the `multicoil` option.

Let's have it only once, and with the `multicoil` option I would say within the nufft function, to keep the status quo

Reply:
We need to apply twice right? Once for the input data and once for the output data?

Reply:
And for output image, its not multicoil really right?

Reply:
```python
        orig_image_channels = ortho_ifft2d(kspace)
        image = adjust_image_size(image, self.image_size)
        nc_kspace = nufft(self.nufft_obj, orig_image_channels, traj, self.image_size, multicoil=self.multicoil)
        nc_kspace, image = scale_tensors(nc_kspace, image, scale_factor=self.scale_factor)
```
`orig_image_channels` is per channel image from the k-space. We resize this with `multicoil` inside `nufft`. `image` is reference output image. This isnt `multicoil`. 

>> Let's have it only once, and with the multicoil option I would say within the nufft function, to keep the status quo

I think we do this for `orig_image_channels` . But we need to make sure same happens for `image`

Reply:
Aaaaah I had indeed read too quickly.
I see now where the confusion is.

Indeed the reference output image is not multicoil. But we don't want to reshape it. The image size thing here is just because of the way tfkbnufft is implemented, which currently requires a fixed image size to be efficient. The `image_size` arguments corresponds to that, which is the in-network image size on which the nufft will operate.

However, in my opinion you are still expected to output an image whose size is the one from the dataset.

Reply:
Well i disagree specifically for Non cartesian case. We dont have any norms yet for that right? 
I dont see a better solution for now than to reshaping all the data. Dont you also do the same in your NCPDNet from what I remember? 
Anyway I think this affects very small percentage of the dataset right?

Reply:
We don't have norms, but I figured that it was best to keep the same practice as for the cartesian case, especially since we don't actually need to have an image whose size is adjusted.

In NCPDNet, like I said above, I do the resizing only for the application of the NUFFT, the rest is left to the original image size.

It affects only a very small percentage of the dataset by cropping it, but for most of the dataset it actually pads it, so it does affect a large part of the dataset overall.

Reply:
But wouldnt this cause issue for loss and other metrics? Due to a shape mismatch i.e.

Reply:
Ahhh cool, now I think I get it, `orig_shape` holds the original shape with which the things can still be applied right

Reply:
But as of now `orig_shape` is shape from `kspace`. Are we sure it matches the shape of results?

`orig_shape = tf.ones([tf.shape(kspace)[0]], dtype=tf.int32) * tf.shape(kspace)[-1]`

Reply:
hmm let's see.
Here I used it only for the knee data. So I basically didn't have a problem with the output shape actually because it was always 320x320. So this shape was purely for in-network convolutions (I am basically using the original image shape in order not to perform unnecessary convolutions, but it might be overengineering honestly).

I think indeed for the brain data it's going to be another story because you need to keep the actual output shape.
Also the output shape for the brain data has to be specified by more than just the phase direction size.

This might indeed need some rewriting of the data pipeline and the ncpdnet to handle brain data.
Basically, what we can do is to just use the output shape, and not do resizing in the network (I think the gain is marginal), only crop/pad at the end but not more.

Do you see my points?

Reply:
Wasnt this what we were doing in first place? Basically have codes to adjust all input images to dimensions we need?

Reply:
thing with brain data is i have seen 320x320 and 384x384 etc... Making all large just makes things really computationally intensive..